packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -217,11 +217,22 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
   TextSelectionOverlay _selectionOverlay;
 
   final ScrollController _scrollController = new ScrollController();
+  bool _didAutoFocus = false;
 
   @override
   void initState() {
     super.initState();
     _currentValue = config.value;
+    config.focusNode.addListener(_handleFocusChanged);
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_didAutoFocus && config.autofocus) {
+      FocusScope.of(context).autofocus(config.focusNode);
+      _didAutoFocus = true;
+    }
```

Comment:
if i globalkey-reparent the EditableText's parent (the one that created the focus node) from one focus scope to another while the editable text is focused, what updates the focus tree?